src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.cs

```diff
@@ -118,18 +146,36 @@ public static new bool Equals(object objA, object objB)
             IThreadingContext threadingContext,
             IGlobalOptionService globalOptions,
             ITextBufferVisibilityTracker? visibilityTracker,
+            TaggerThreadCoordinator? threadCoordinator,
             IAsynchronousOperationListener asyncListener)
         {
             ThreadingContext = threadingContext;
             GlobalOptions = globalOptions;
             AsyncListener = asyncListener;
             _visibilityTracker = visibilityTracker;
+            _threadCoordinator = threadCoordinator;
 
 #if DEBUG
             StackTrace = new StackTrace().ToString();
 #endif
         }
 
+        private async Task PerformUIWorkAsync(Func<CancellationToken, Task> work, CancellationToken cancellationToken)
+        {
+            if (_threadCoordinator is null)
```

Comment:
this is the TS case until they update their side of things.  Once they do, this field will become non-optional.